src/webparts/sendToLaserficheLoginComponent/components/SendToLaserficheLoginComponent.tsx

```diff
@@ -11,641 +9,306 @@ import {
-        const Entryid9 = entryCreateResult.operations.entryCreate.entryId;
-        filelink = `https://app.${this.state.region}/laserfiche/DocView.aspx?db=${repoId}&docid=${Entryid9}`;
-
-        document.getElementById('it').innerHTML = 'Document uploaded';
-        document.getElementById('imgid').style.display = 'none';
-        document.getElementById('divid').style.display = 'block';
-        document.getElementById('divid1').onclick = this.Dc;
-        document.getElementById('divid13').style.display = 'block';
-        document.getElementById('divid13').onclick = this.viewfile;
-        document.getElementById('divid14').onclick = this.Redirect;
-        if (Action === ActionTypes.COPY) {
-          window.localStorage.removeItem(TempStorageKeys.LContType);
-        } else if (Action === ActionTypes.MOVE_AND_DELETE) {
-          this.DeleteFile(pageOrigin, Fileurl, Filenamewithext);
-        } else if (Action === ActionTypes.REPLACE) {
-          this.deletefileandreplace(
-            pageOrigin,
-            Fileurl,
-            Filenamewithoutext,
-            Filenamewithext,
-            filelink,
-            contextPageAbsoluteUrl
-          );
-        } else {
-          // TODO
-        }
-      } catch (error) {
-        if (error.operations.setFields.exceptions[0].statusCode === 409) {
-          const entryidConflict5 = error.operations.entryCreate.entryId;
-          filelink = `https://app.${this.state.region}/laserfiche/DocView.aspx?db=${repoId}&docid=${entryidConflict5}`;
-
-          document.getElementById('it').innerHTML =
-            'Document uploaded to repository, updating metadata failed due to constraint mismatch<br/> <p style="color:red">The Laserfiche template and fields were not applied to this document</p>';
-          document.getElementById('imgid').style.display = 'none';
-          document.getElementById('divid').style.display = 'block';
-          document.getElementById('divid1').onclick = this.Dc;
-          document.getElementById('divid13').style.display = 'block';
-          document.getElementById('divid13').onclick = this.viewfile;
-          document.getElementById('divid14').onclick = this.Redirect;
-          window.localStorage.removeItem(TempStorageKeys.LContType);
-        } else {
-          window.alert(`Error uploding file: ${JSON.stringify(error)}`);
-          window.localStorage.removeItem(TempStorageKeys.LContType);
-          dialog.close();
-        }
+    } catch (error) {
+      const conflict409 =
+        error.operations.setFields.exceptions[0].statusCode === 409;
+      if (conflict409) {
+        const entryidConflict1 = error.operations.entryCreate.entryId;
+
+        const fileLink = getEntryWebAccessUrl(
+          entryidConflict1.toString(),
+          repoId,
+          webClientUrl,
+          false
+        );
+        const pageOrigin = spFileMetadata.pageOrigin;
+        const fileUrl = spFileMetadata.fileUrl;
```

Comment:
does 409 means that the document is failed to be saved to laserfiche? how is this dialog different than a successful dialog?

Reply:
409 is a conflict, and the API team uses it to say that the entry was saved without metadata, but this should be reviewed